pygmt/tests/test_grd2xyz.py

```diff
@@ -0,0 +1,80 @@
+"""
+Tests for grd2xyz.
+"""
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+from pygmt import grd2xyz
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(resolution="01d", region=[-1, 1, -1, 1])
+
+
+def test_grd2xyz(grid):
+    """
+    Make sure grd2xyz works as expected.
+    """
+    xyz_data = grd2xyz(grid=grid, output_type="numpy")
+    assert xyz_data.shape == (4, 3)
+
+
+def test_grd2xyz_format(grid):
+    """
+    Test that correct formats are returned.
+    """
+    xyz_default = grd2xyz(grid=grid)
+    assert isinstance(xyz_default, pd.DataFrame)
+    assert list(xyz_default.columns) == ["lon", "lat", "elevation"]
```

Comment:
```suggestion
    lon = -0.5
    lat = 3.5
    orig_val = grid.sel(lon=lon, lat=lat).to_numpy()
    xyz_default = grd2xyz(grid=grid)
    xyz_val = xyz_default[(xyz_default["lon"] == lon) & (xyz_default["lat"] == lat)][
        "elevation"
    ].to_numpy()
    assert isinstance(xyz_default, pd.DataFrame)
    assert orig_val.size == 1
    assert xyz_val.size == 1
    np.testing.assert_allclose(orig_val, xyz_val)
```
Here is an example that would work for testing Wei Ji's suggestion. The test currently fails because 'lon' and 'lat' are switched for the output DataFrame.